### PR TITLE
[PiranhaJava] Deleting statements containing specific Mockito API patterns 

### PR DESCRIPTION
**Pull request by ketkarameya**
* This addresses the (i) of issue #104  
* Made use of the Matchers API (the mini predicate DSL) provided by Error prone 
* Currently I have added support only for the Mockito's `when(...).then*()` pattern.

### CODE REVIEW

File: java/piranha/src/test/java/com/uber/piranha/CorePiranhaTest.java [hunk 1]
```diff
@@ -499,4 +499,64 @@ public void testIgnoresPrefixMatchFlag() throws IOException {
...
+            " private XPTest experimentation;",
+            " public void evaluate() {",
+            "  Answer ans = new Answer<Integer>() {\n"
+                + "      public Integer answer(InvocationOnMock invocation) throws Throwable {\n"
+                + "        return (Integer) invocation.getArguments()[0];\n"
+                + "      }};",
+            "  when(experimentation.isToggleDisabled(\"STALE_FLAG\")).thenReturn(false);",
+            "  when(experimentation.isToggleEnabled(\"STALE_FLAG\")).thenThrow(new RuntimeException());",
+            "  when(experimentation.isToggleDisabled(\"STALE_FLAG\")).thenCallRealMethod();",
+            "  when(experimentation.isToggleEnabled(\"STALE_FLAG\")).then(ans);",
```
Comment (mkr-plse) [inline java/piranha/src/test/java/com/uber/piranha/CorePiranhaTest.java:532]: can we also add a non stale flag in the mix?

File: java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java [hunk 2]
```diff
@@ -820,6 +852,14 @@ public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState
     }
 
     Value x = evalExpr(tree, state);
+    if (x == Value.TRUE || x == Value.FALSE) {
+      if (enclosingNode(MOCKITO_UNNECESSARY_MOCKING_PATTERN).matches(tree, state)) {
+        StatementTree stmt = ASTHelpers.findEnclosingNode(state.getPath(), StatementTree.class);
+        if (stmt != null) {
+          return buildDescription(stmt).addFix(SuggestedFix.delete(stmt)).build();
```
Comment (mkr-plse) [inline java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java:859]: can the changes happen in updateCode? If not, maybe update the `startPos` and `endPos` here?

File: java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java [hunk 3]
```diff
@@ -695,13 +704,16 @@ private Description updateCode(
     }
 
     if (update) {
-      Description.Builder builder = buildDescription(tree);
-      SuggestedFix.Builder fixBuilder = SuggestedFix.builder();
-      fixBuilder.replace(expr, replacementString);
+      SuggestedFix.Builder fixBuilder =
+          Optional.ofNullable(
+                  ASTHelpers.findEnclosingNode(state.getPath(), ExpressionStatementTree.class))
+              .filter(
```
Comment (ketkarameya) [inline java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java:710]: When we'll have more such API patterns to support we can extract out these lines.

**Review by mkr-plse — APPROVED** [review]
lgtm. thanks for the PR.

**Pull request merged**